### PR TITLE
[LAP-114] [LAP-126] Trigger Events with Animations

### PR DESCRIPTION
**Pull request by Algo-ryth-mix**
Allows for the Animator and AnimationEditor to configure & trigger custom events via the eventbus through animations.
A new Editor GUI was added to make configuration easy.

Implements [LAP-114] and [LAP-126]

[LAP-114]: https://legion-engine.atlassian.net/browse/LAP-114
[LAP-126]: https://legion-engine.atlassian.net/browse/LAP-126
grafik

### CODE REVIEW

File: applications/sandbox/data/animation.cpp [hunk 1]
```diff
@@ -14,6 +14,12 @@ namespace ext
         {
             return core::math::quat{ a[0].get<float>(), a[1].get<float>(), a[2].get<float>(), a[3].get<float>() };
         }
+        std::unordered_map<std::string_view,id_type> g_AnimationEventDatabase =
+            {std::make_pair(std::string_view("VoidAnimationEvent"),typeHash<void_animation_event>())};
+        std::unordered_map<id_type,std::string_view> g_ReverseAnimationEventDatabase =
+            {std::make_pair(typeHash<void_animation_event>(),std::string_view("VoidAnimationEvent"))};
```
Comment (GlynLeine) [inline applications/sandbox/data/animation.cpp:20]: is this a default event?

**Review by GlynLeine — APPROVED** [review]

File: legion/engine/audio/data/importers/audio_importers.cpp [hunk 2]
```diff
@@ -74,6 +74,9 @@ namespace legion::audio
         using common::Err, common::Ok;
         using decay = common::result_decay_more<audio_segment, fs_error>;
 
+        log::debug("Attempting to load WAV file");
+
+
         RIFF_Header header;
```
Comment (Jelled1st) [inline legion/engine/audio/data/importers/audio_importers.cpp]: This can probably be removed

**Review by Jelled1st — APPROVED** [review]

File: applications/sandbox/data/animation.cpp (hunk 1 above)
Comment (Algo-ryth-mix) [reply applications/sandbox/data/animation.cpp:20]: yes

File: legion/engine/audio/data/importers/audio_importers.cpp (hunk 2 above)
Comment (Algo-ryth-mix) [reply legion/engine/audio/data/importers/audio_importers.cpp]: it probably could yes

**Pull request merged**